src/java/org/apache/cassandra/utils/JVMStabilityInspector.java

```diff
@@ -140,15 +140,26 @@ else if (t instanceof UnrecoverableIllegalStateException)
             if (t instanceof FSError || t instanceof CorruptSSTableException)
                 isUnstable = true;
 
-        fn.accept(t);
-
         // Check for file handle exhaustion
         if (t instanceof FileNotFoundException || t instanceof FileSystemException || t instanceof SocketException)
             if (t.getMessage() != null && t.getMessage().contains("Too many open files"))
                 isUnstable = true;
 
         if (isUnstable)
+        {
+            if (!StorageService.instance.isDaemonSetupCompleted())
```

Comment:
nit: Not even in this diff, but the body of `isDaemonSetupCompleted()` could just be...

```
return daemon != null && daemon.setupCompleted();
```